File: P12926.py
def solution(s, n):
    answer = ''
    for a in s:
        if ord('a') <= ord(a) <= ord('z'):
            answer += chr((ord(a) - ord('a') + n) % 26 + ord('a'))
        elif ord('A') <= ord(a) <= ord('Z'):
            answer += chr((ord(a) - ord('A') + n) % 26 + ord('A'))
        else:
            answer += " "
    return answer

File: P12926.java
public class P12926 {
    public String solution(String s, int n) {
        String answer = "";
        for (int i=0; i<s.length(); i++) {
            char digit = s.charAt(i);
            if ('a' <= digit && digit <= 'z') {
                digit -= 'a';
                digit += n;
                digit %= 26;
                digit += 'a';
            }
            else if ('A' <= digit && digit <= 'Z') {
                digit -= 'A';
                digit += n;
                digit %= 26;
                digit += 'A';
            }
            answer += digit;
        }
        return answer;
    }
}